src/generate/generic.rs

```diff
@@ -227,3 +227,63 @@ pub enum Variant<U, T> {
     ///Raw bits
     Res(U),
 }
+
+use Variant::*;
+impl<U, T> Variant<U, T> {
+    /// Check if the variant is expected
+    pub fn is_value(&self) -> bool {
+        match self {
+            Val(_) => true,
+            Res(_) => false,
+        }
+    }
+    /// Check if the variant is not expected
+    pub fn is_reserved(&self) -> bool {
+        match self {
+            Val(_) => false,
+            Res(_) => true,
+        }
+    }
+
+    /// Returns the contained value or a default
+    #[inline]
+    pub fn unwrap_or(self, def: T) -> T {
+        match self {
+            Val(v) => v,
+            Res(_) => def,
+        }
+    }
+
+    /// Returns the contained value or computes it from a closure
+    #[inline]
+    pub fn unwrap_or_else<F: FnOnce(U) -> T>(self, f: F) -> T {
+        match self {
+            Val(v) => v,
+            Res(u) => f(u),
+        }
+    }
+}
+
+impl<U, T> Variant<U, T> where U: core::fmt::Debug {
+    /// Moves the value `v` out of the `Variant` if it is `Val(v)`.
+    ///
+    /// Panics if the self value equals `Res`
+    #[inline]
+    pub fn unwrap(self) -> T {
+        match self {
+            Val(v) => v,
+            Res(u) => panic!("Unexpected variant: {:?}", u),
```

Comment:
I'm not to hot about this `panic!` and the `expect` method below. If this ends up in use in real code it'd add significant bloat.

Reply:
I can remove `unwrap` and keep `expect` as it is more explicit.

Reply:
I'm fine with with those methods per se but let's get rid of the formatting and the `Debug` bound, please.

Reply:
Ok

Reply:
Is now better?